src/odemis/gui/comp/buttons.py

```diff
@@ -687,6 +688,71 @@ def OnLeftDown(self, event):
         self.Refresh()
 
 
+class ProgressRadioButton(GraphicRadioButton):
+    """
+    Button that when pressed down, first goes to a "in progress" image, but
+    can also be requested to show a "completed" image.
+    Instead of having value False/True, it has 0 (untoggled), 1 (toggled and in progress),
+    2 (toggled and completed).
+    There is no way for the user to explicitly set the button to completed, this
+    can only be done programmatically.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        icon: (wx.Bitmap) when untoggled
+        icon_progress: (wx.Bitmap) when toggled and in progress
+        icon_on: (wx.Bitmap) when toggled and completed
+        (wx.Bitmap)
+        """
+        self._icon_progress = kwargs.pop('icon_progress', None)
+
+        # The super class has a .icon_on, which we use to show both the progress and on icons
+        self._actual_icon_on = kwargs.get('icon_on', None)
+        self._completed = True
+        super().__init__(*args, **kwargs)
+
+    def GetValue(self):
+        if self.up:
+            if self._completed:
+                return 2
+            else:
+                return 1
+        else:
+            return 0
```

Comment:
The SetValue shows the functioning of return values 0,1 and 2. Not sure if the functioning of 0,1 and 2 is defined somewhere. Otherwise may be we can add comments here.
```suggestion
    def GetValue(self):
        if self.up:
            if self._completed:
                # icon turns to green to mark the completion 
                return 2
            else:
                # icon stays/becomes orange during the work in progress
                return 1
        else:
            # icon ...
            return 0
```